tests/skylab/hca_adapter/pr/ValidateHcaAdapter.wdl

```diff
@@ -0,0 +1,328 @@
+version 1.0
+
+task ValidateOptimusDescriptorAnalysisFiles {
+  input {
+    File optimus_descriptors_analysis_file_intermediate_loom_json
+    File optimus_descriptors_analysis_file_intermediate_loom_json_truth
+    File optimus_descriptors_analysis_file_intermediate_bam_json
+    File optimus_descriptors_analysis_file_intermediate_bam_json_truth
+    File optimus_descriptors_analysis_file_intermediate_reference_json
+    File optimus_descriptors_analysis_file_intermediate_reference_json_truth
+    }
+
+ command <<<
+       # catch intermittent failures
+       set -eo pipefail
+
+       #testing descriptors/analsysis_file/.loom
+       diff "~{optimus_descriptors_analysis_file_intermediate_loom_json}" "~{optimus_descriptors_analysis_file_intermediate_loom_json_truth}"
+
+         if [ $? -ne 0 ];
+         then
+          echo "error"
+         fi
+
+       #testing descriptors/analsysis_file/.bam
+       diff "~{optimus_descriptors_analysis_file_intermediate_bam_json}" "~{optimus_descriptors_analysis_file_intermediate_bam_json_truth}"
+
+                if [ $? -ne 0 ];
+                then
+                 echo "error"
+                fi
+
+       #testing descriptors/reference_file/.fasta
+       diff "~{optimus_descriptors_analysis_file_intermediate_reference_json}" "~{optimus_descriptors_analysis_file_intermediate_reference_json_truth}"
+
+                if [ $? -ne 0 ];
+                then
+                 echo "error"
+                fi
+  >>>
+
+  runtime {
+    docker: "quay.io/humancellatlas/secondary-analysis-samtools:v0.2.2-1.6"
+    cpu: 1
+    memory: "3.75 GiB"
+    disks: "local-disk 10 HDD"
+  }
+}
+
+task ValidateOptimusLinksFiles {
```

Comment:
It looks like these are all fundamentally doing the same thing. Consider collapsing into a single task that compares 2 files and then calling that task on different outputs:

```
task CompareAdapterFiles {
  input {
    File test_json
    FIle truth_json
  }
  command <<<
  set -eo pipefail
  diff "~{test_json}" "~{truth_json}"

  if [ $? -ne 0 ];
  then
   echo "Error: ${test_json} and ${truth_json} differ"
  fi
  >>>

  runtime {
    docker: "quay.io/humancellatlas/secondary-analysis-samtools:v0.2.2-1.6"
    cpu: 1
    memory: "3.75 GiB"
    disks: "local-disk 10 HDD"
  }
}
```

Then to call it as different tasks in the other WDL (so you can see quickly which one is failing) it would look something like:

```
call checker_adapter.CompareAdapterFiles as checker_adapter_metadata_analysis_process {
    input:
      test_json=target_adapter.analysis_process,
      truth_json=optimus_metadata_analysis_process_file_intermediate_json
  }
```